packages/teams-js/src/internal/videoPerformanceStatistics.ts

```diff
@@ -0,0 +1,114 @@
+import { VideoFrameTick } from './videoFrameTick';
+
+export type VideoPerformanceStatisticsResult = {
+  effectId: string;
+  frameWidth: number;
+  frameHeight: number;
+  duration: number; // the duration in milliseconds that the data were collected
+  sampleCount: number; // the number of frames that were processed in the duration
+  distributionBins: Uint32Array; // an array that presents counts of frames that were finished in n milliseconds: distributionBins[frameProcessingDurationInMs]=frameCount. For example, distributionBins[10] = 5 means that 5 frames were processed in 10 milliseconds.
+};
+
+export class VideoPerformanceStatistics {
+  private static readonly initialSessionTimeoutInMs = 1000;
+  private static readonly maxSessionTimeoutInMs = 1000 * 30;
+
+  private currentSession: {
+    startedAtInMs: number;
+    timeoutInMs: number;
+    effectId: string;
+    frameWidth: number;
+    frameHeight: number;
+  };
+
+  private frameProcessingStartedAt: number;
+  private distributionBins: Uint32Array;
+  private sampleCount = 0;
+  private timeoutId: string;
+
+  public constructor(
+    distributionBinSize: number,
+    private report: (result: VideoPerformanceStatisticsResult) => void, // post event to the host
+  ) {
+    this.distributionBins = new Uint32Array(distributionBinSize);
+  }
+
+  /**
+   * Call this function before processing every frame
+   */
+  public processStarts(effectId: string, frameWidth: number, frameHeight: number): void {
```

Comment:
The comment here says "call this function before processing every frame". Should there be any validation that it was indeed called before processing frames occurs?

Reply:
There is no way in this class to know if this method is called before processing a frame, I added a UT in videoPerformanceMonitor.spcc.ts to ensure this.